все працює

DimaYermolenkoOut opened the pull request:
створив нову гілку

створив новий venv

pip install requarements.txt

додатково pip install factory_boy

створти базу Postgreql - я це зробив вручну в PgAdmin

потім 

python manage.py makemigrations products
python manage.py migrate  

генерую моделі

python manage.py generateproducts   


створив модель Store , StoreInventory в папці products/models
додав   'Store' , 'StoreInventory' в  __init__   в папці models

python manage.py makemigrations products
python manage.py migrate

в admin.py додав таблиці в адмінку
admin.site.register(StoreInventory)  
admin.site.register(Store)

щоб перевірити як вони відображаються створив superuser

python manage.py createsuperuser
Yermolenko
admin

виникла проблема що в pycharm нові моделі не з'явилися і щось сварилося на сервер. 
запустив pgAdmin на компі і все спрацювало (може тому що цю частину робив наступного дня і сервер postgresql  не був запущений)


далі робимо Api

створюємо серіалізатори на Store і StoreInventory



потім розширюємо viewsets.py
```python
class StoreViewSet(viewsets.ModelViewSet):  
    queryset = Store.objects.all()  
    serializer_class = StoreSerializer  
    permission_classes = [IsAuthenticatedOrReadOnly]  
  
  
class StoreInventoryViewSet(viewsets.ModelViewSet):  
    queryset = StoreInventory.objects.all()  
    serializer_class = StoreInventorySerializer  
    permission_classes = [IsAuthenticatedOrReadOnly]
```

в urls.py
```python
# додав рядочки
router.register('stores', StoreViewSet)  
router.register('store-inventories', StoreInventoryViewSet)
```

json StoreInventorySerializer був тільки з id без name і це не зручно (бо це спільна таблиця many-to-many)
тому вручну дописуємо product_id product_name store_id store_name

```python
class StoreSerializer(serializers.ModelSerializer):  
    class Meta:  
        model = Store  
        fields = ['id', 'name', 'address', 'description', 'email', 'phone']


class StoreInventorySerializer(serializers.ModelSerializer):  
    product_id = serializers.PrimaryKeyRelatedField(source='product.id', queryset=Product.objects.all())  
    product_name = serializers.CharField(source='product.title', read_only=True)  
    store_id = serializers.PrimaryKeyRelatedField(source='store.id', queryset=Store.objects.all())  
    store_name = serializers.CharField(source='store.name', read_only=True)  
    class Meta:  
        model = StoreInventory  
        fields = ['id', 'store_id', 'store_name', 'product_id','product_name', 'quantity', 'price']
```

http://localhost:8000/api/stores/
[
    {
        "id": 1,
        "name": "Store_1_Y",
        "address": "Kyiv, One str, 1",
        "description": "магазин продуктів харчування",
        "email": "[store_1_y@django.com](mailto:store_1_y@django.com)",
        "phone": "+380631111111"
    },
    {
        "id": 2,
        "name": "Store_2_Y",
        "address": "Kyiv, Two str, 2",
        "description": "магазин продуктів харчування і пекарня",
        "email": "[store_2_y@django.com](mailto:store_2_y@django.com)",
        "phone": "+380632222222"
    },
    {
        "id": 3,
        "name": "Store_3_Y",
        "address": "Kyiv, Three str, 3",
        "description": "магазин іграшок",
        "email": "[store_3_y@django.com](mailto:store_3_y@django.com)",
        "phone": "+380633333333"
    }
]

http://localhost:8000/api/store-inventories/

[
    {
        "id": 1,
        "store_id": 1,
        "store_name": "Store_1_Y",
        "product_id": 2,
        "product_name": "why",
        "quantity": 50.0,
        "price": "50.00"
    },
    {
        "id": 2,
        "store_id": 2,
        "store_name": "Store_2_Y",
        "product_id": 3,
        "product_name": "four",
        "quantity": 150.0,
        "price": "150.00"
    },
    {
        "id": 3,
        "store_id": 3,
        "store_name": "Store_3_Y",
        "product_id": 6,
        "product_name": "international",
        "quantity": 100.0,
        "price": "100.00"
    }
]

через Postman не зміг, не зрозумів про авторизацію, все працює в на сторінці Django REST framework

DimaYermolenkoOut committed:
все працює

DimaYermolenkoOut committed:
add paginations

DimaYermolenkoOut committed:
Update settings.py

test work on gitHub 

paginations 8

DimaYermolenkoOut committed:
add paginations

p1v2 commented on products/models/product.py:
```diff
@@ -18,5 +18,8 @@ class Product(models.Model):
 
+    create_at = models.DateTimeField(auto_now_add=True)
+    update_at = models.DateTimeField(auto_now=True)
+
```
pls add relatiton to the store through model StoreInventory as we did in the lesson between order and product.

p1v2 commented on products/models/store.py:
```diff
@@ -0,0 +1,12 @@
+    description = models.TextField(blank=True, null=True)
+    email = models.EmailField(blank=True, null=True)
+    phone = models.CharField(max_length=20, blank=True, null=True)
+
```
same here, pls add relation to the product through model StoreInventory as we did in the lesson between order and product.

DimaYermolenkoOut committed:
Merge remote-tracking branch 'p1v2/main' into Yermolenko---dz-7

# Conflicts:
#	hillelDjango3/settings.py
#	hillelDjango3/urls.py
#	products/filters.py
#	products/management/commands/generateproducts.py
#	products/models/product.py
#	products/serializers/product.py
#	products/viewsets.py
#	requirements.txt

DimaYermolenkoOut committed:
21-02-24
налаштування після оновлення

DimaYermolenkoOut committed:
не запусколось
сварилось на created_at
змінив на create_at
в filters.py і viewsets.py / latest
працює

DimaYermolenkoOut committed:
add top_selling_products

DimaYermolenkoOut committed:
23-02-2024
redis work
task doest work

DimaYermolenkoOut committed:
24-02-2024
redis work
task  work in shell
in flower i can see task
but url doesn't work

DimaYermolenkoOut committed:
24-02-2024
є питання
редагував views
простий запит до бази працює
через task не зрозуміло

DimaYermolenkoOut committed:
25-02-2024

p1v2 commented:
Давайте на уроці розберемо ваші питання. По коду все правильно

DimaYermolenkoOut committed:
26-02-2024
на уроці
